src/app/entity/entity.ts

```diff
@@ -23,25 +23,47 @@
  */
 export class Entity {
 
-  private _id: string;
+  /**
+   * The unique id of this entity.
+   */
+  private readonly entityId: string;
 
-  constructor(id: string) {
-    this.setId(id);
-  }
+  /**
+   * The entity's type.
+   */
+  private readonly type: string;
 
-
-  public getPrefix(): string {
-    return '';
+  /**
+   * Creates an entity object with the given id. This id is final and won't be changeable after this object has been
+   * created.
+   *
+   * @param id a unique id for this entity
+   */
+  constructor(id: string) {
+    this.entityId = id;
+    this.type = this.constructor.name;
```

Comment:
What do inheriting classes like `User` have to do to ensure `type` is set correctly?

Do they have to explicitly call the super constructor? (and is `this.constructor.name` then really the name of the child class?)
Or do they have to include this line into their constructor(s) explicitly?